pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercuryOrderDao.java

```diff
@@ -121,69 +132,74 @@ public static void updateOrderStatus(BaseMercuryStatusMessage baseMercuryStatusM
         MercuryOrderUseCase.exportStatusToES(baseMercuryStatusMessage, clinicalOrderAtomicReference.get(), statusDate);
     }
 
-    public HashMap<String, MercuryOrderDto> getPossibleBarcodesForParticipant(String ddpParticipantId) {
+    public Map<String, MercuryOrderDto> getPossibleBarcodesForParticipant(String ddpParticipantId) {
         HashMap<String, MercuryOrderDto> map = new HashMap<>();
         SimpleResult results = inTransaction((conn) -> {
             SimpleResult dbVals = new SimpleResult();
             try (PreparedStatement stmt = conn.prepareStatement(SQL_GET_KIT_FROM_BARCODE_KIT_LABEL)) {
                 stmt.setString(1, ddpParticipantId);
-                try (ResultSet rs = stmt.executeQuery()) {
-                    while (rs.next()) {
-                        MercuryOrderDto mercuryOrderDto = new MercuryOrderDto(rs.getString(DBConstants.DDP_PARTICIPANT_ID),
-                                rs.getString(DBConstants.MERCURY_ORDER_CREATOR), rs.getString(DBConstants.KIT_LABEL),
-                                rs.getInt("ktype." + DBConstants.KIT_TYPE_ID), rs.getInt(DBConstants.DDP_INSTANCE_ID),
-                                null, rs.getLong("kit." + DBConstants.DSM_KIT_REQUEST_ID));
-                        log.info("found related info about barcode " + mercuryOrderDto.getBarcode());
-                        map.put(mercuryOrderDto.getBarcode(), mercuryOrderDto);
-                    }
-                } catch (Exception e) {
-                    throw new RuntimeException("Error getting possible mercury orders for participant " + ddpParticipantId, e);
-                }
-            } catch (SQLException ex) {
+                getKitsFromStatement(stmt, map);
+            } catch (Exception ex) {
```

Comment:
I think we still want to catch SQLException here, not Exception. This may just lead to stacking DsmInternalErrors on top of each other.